packages/schema/src/types/discriminatedObject.ts

```diff
@@ -39,43 +45,64 @@ export function discriminatedObject<
         return discriminatorMap[discriminatorValue];
       }
     }
+    return undefined;
+  };
+  const allSchemas = Object.values(discriminatorMap).reverse();
+  const selectSchema = (
+    value: unknown,
+    discriminatorProp: string | TDiscrimProp | TDiscrimMappedProp,
+    checker: (schema: TSchema) => SchemaValidationError[],
```

Comment:
should we rename "checker" to "validator"

Reply:
yes we should!